crates/walrus-sdk/src/client/quilt_client.rs

```diff
@@ -0,0 +1,583 @@
+                certified_epoch,
+                Self::MAX_RETRIEVE_SLIVERS_ATTEMPTS,
+                Self::TIMEOUT_DURATION,
+            )
+            .await?;
+
+        let first_sliver = slivers.first().expect("the first sliver should exist");
+        let quilt_version = QuiltVersionEnum::new_from_sliver(first_sliver.symbols.data())?;
+
+        let quilt_index = match quilt_version {
+            QuiltVersionEnum::V1 => {
+                self.retrieve_quilt_index_internal::<QuiltVersionV1>(
+                    metadata,
+                    certified_epoch,
+                    &slivers,
+                )
+                .await?
+            }
+        };
+
+        Ok(quilt_index)
+    }
+
+    async fn retrieve_quilt_index_internal<V: QuiltVersion>(
+        &self,
+        metadata: &VerifiedBlobMetadataWithId,
+        certified_epoch: Epoch,
+        first_sliver: &[SliverData<V::SliverAxis>],
+    ) -> ClientResult<QuiltIndex>
+    where
+        SliverData<V::SliverAxis>: TryFrom<Sliver>,
+    {
+        let mut all_slivers = Vec::new();
+        let mut refs = Vec::new();
+        let first_sliver_refs: Vec<_> = first_sliver.iter().collect();
+        let mut decoder = V::QuiltConfig::get_decoder(&first_sliver_refs);
+
+        let quilt_index = match decoder.get_or_decode_quilt_index() {
+            Ok(quilt_index) => quilt_index,
+            Err(QuiltError::MissingSlivers(indices)) => {
+                all_slivers.extend(
+                    self.client
+                        .retrieve_slivers_with_retry::<V::SliverAxis>(
+                            metadata,
+                            &indices,
+                            certified_epoch,
+                            Self::MAX_RETRIEVE_SLIVERS_ATTEMPTS,
+                            Self::TIMEOUT_DURATION,
+                        )
+                        .await?,
+                );
+                refs.extend(all_slivers.iter());
+                decoder.add_slivers(&refs);
+                decoder.get_or_decode_quilt_index()?
+            }
+            Err(e) => return Err(e.into()),
+        };
+
+        Ok(quilt_index)
+    }
+
+    /// Retrieves the quilt patches of the given identifiers from the quilt.
+    pub async fn get_blobs_by_identifiers(
+        &self,
+        quilt_id: &BlobId,
+        identifiers: &[&str],
+    ) -> ClientResult<Vec<QuiltStoreBlobOwned>> {
+        let metadata = self.get_quilt_metadata(quilt_id).await?;
+
+        let blobs = match metadata {
+            QuiltMetadata::V1(metadata) => {
+                self.get_blobs_by_identifiers_impl::<QuiltVersionV1>(
+                    &metadata.get_verified_metadata(),
+                    &metadata.index.into(),
+                    identifiers,
+                )
+                .await?
+            }
+        };
+
+        Ok(blobs)
+    }
+
+    /// Retrieves quilt patches from QuiltV1 by identifiers.
+    async fn get_blobs_by_identifiers_impl<V: QuiltVersion>(
```

Comment:
Question: Is this function specific to V1 as the docstring indicates or generic as the function signature?